geode-core/src/test/java/org/apache/geode/internal/cache/InternalCacheBuilderTest.java

```diff
@@ -101,109 +81,72 @@ public void setUp() {
+        new Properties(), null, metricsSessionBuilder, nullSingletonSystemSupplier,
+        constructorOf(constructedSystem()),
         nullSingletonCacheSupplier, constructorOf(constructedCache()));
 
-    Throwable thrown = catchThrowable(() -> internalCacheBuilder
-        .create());
-
-    assertThat(thrown).isInstanceOf(NullPointerException.class);
+    assertThatThrownBy(internalCacheBuilder::create)
+        .isInstanceOf(NullPointerException.class);
   }
 
   @Test
   public void create_constructsSystem_withGivenProperties_ifNoSystemExists_andNoCacheExists() {
-    InternalCache constructedCache = constructedCache();
-
     InternalDistributedSystemConstructor systemConstructor = constructorOf(constructedSystem());
     Properties configProperties = new Properties();
 
     InternalCacheBuilder internalCacheBuilder = new InternalCacheBuilder(
-        configProperties, new CacheConfig(),
-        compositeMeterRegistryFactory, metricsSessionInitializer,
-        nullSingletonSystemSupplier, systemConstructor,
-        nullSingletonCacheSupplier, constructorOf(constructedCache));
-
-    internalCacheBuilder
-        .create();
-
-    verify(systemConstructor).construct(same(configProperties), any());
-  }
-
-  @Test
-  public void create_constructsCompositeMeterRegistry_ifNoCacheExists() {
-    int theSystemId = 21;
-    String theMemberName = "theMemberName";
-    String theHostName = "theHostName";
-    boolean isClient = false;
-    InternalDistributedSystem theConstructedSystem =
-        systemWith("theConstructedSystem", theSystemId, theMemberName, theHostName);
-    InternalCache constructedCache = constructedCache();
-
-    CompositeMeterRegistryFactory theCompositeMeterRegistryFactory =
-        mock(CompositeMeterRegistryFactory.class);
-
-    InternalCacheBuilder internalCacheBuilder = new InternalCacheBuilder(
-        new Properties(), new CacheConfig(),
-        theCompositeMeterRegistryFactory, metricsSessionInitializer,
-        nullSingletonSystemSupplier, constructorOf(theConstructedSystem),
-        nullSingletonCacheSupplier, constructorOf(constructedCache));
+        configProperties, new CacheConfig(), metricsSessionBuilder, nullSingletonSystemSupplier,
+        systemConstructor,
+        nullSingletonCacheSupplier, constructorOf(constructedCache()));
 
     internalCacheBuilder
         .create();
 
-    verify(theCompositeMeterRegistryFactory)
-        .create(eq(theSystemId), eq(theMemberName), eq(theHostName), eq(isClient));
+    verify(systemConstructor).construct(same(configProperties), any(), any());
   }
 
   @Test
-  public void create_setsConstructedCompositeMeterRegistry_onTheConstructedCache_ifNoCacheExists() {
-    InternalCache constructedCache = constructedCache();
-    InternalCacheConstructor cacheConstructor = constructorOf(constructedCache);
+  public void create_passesSeparatelyAddedUserRegistriesToSystemConstructor_ifNoSystemExists() {
```

Comment:
I'm a little confused about the name of this test — I'm not sure it's testing what it says it's testing. Should it also verify that `addPersistentMeterRegistry` was called on the metrics service builder for each of the user registries? Or maybe there should be a separate test to verify that?

Reply:
Fixed in a subsequent commit.